### DISCUSSION TITLE
Session7中门电路转化问题

### DISCUSSION THREAD

**Discussion by miles-six** [zkp-co-learning/zkp-co-learn#65] [Q&A（提问题在隔壁~）] [answered]
在session7中，在电路中有意将加法门转化成了乘法门，这样转化的原因是什么呢

![image](https://user-images.githubusercontent.com/66052478/225347152-d076df96-38da-43b7-8af1-331f6efd5b55.png)

**Comment by crazysnut** [comment, marked as answer]
我的理解是所有的门需要转化为R1CS的形式，即
$$L*R=O$$
所以对于加法门，可以有两种处理方式

1. 如问题中图片所示，直接*1或者1*，看变量是左输入还是右输入
2. 如果上级有乘法门，加法门也可以被吸收（优化）
对于Tonglai老师上课的例子，gate2和gate3也可以被优化成：
$$w1*(w4+1)=w6$$
